keps/sig-auth/3257-trust-anchor-sets/README.md

```diff
@@ -0,0 +1,1004 @@
+//
+// A TrustAnchorSets object wraps up multiple trust anchor sets, each of
+// which in turn is a set of one or more trust anchors (individual
+// certificates).  Most use cases will only require a single trust
+// anchor set, conventionally named `default`.  Multiple uniquely-named trust
+// anchor sets will likely only be required in mutual TLS use cases with
+// federation between multiple meshes.
+type TrustAnchorSets struct {
+  metav1.TypeMeta   `json:",inline"`
+  metav1.ObjectMeta `json:"metadata,omitempty"`
+
+  Spec TrustAnchorSetsSpec `json:"spec"`
+  Status TrustAnchorSetsStatus `json:"status"`
+}
+
+type TrustAnchorSetsSpec struct {
+  // The name of the signer that owns this TrustAnchorSets object.
+  //
+  // Each signer may maintain multiple TrustAnchorSets objects.
+  SignerName string
+}
+
+type TrustAnchorSetsStatus struct {
+  // The list of trust anchor sets.
+  Sets []TrustAnchorSet
+
+  // The current conditions of the TrustAnchorSets object.
+  //
+  // There is one canonical condition, "Ready", that the signer uses
+  // to indicate to kubelet that the TrustAnchorSets object is up to date.
+  // This is used to keep pods from starting up with empty trust anchor sets
+  // while the signer is initializing.
+  Conditions []metav1.Condition
+}
+
+type TrustAnchorSet struct {
+  // The unique name of this trust anchor set.
+  Name string
+
+  // The individual trust anchors for this set.  Each entry is a single
+  // PEM-wrapped, DER-formatted X.509 certificate.
+  PEMTrustAnchors []string
+}
+```
+
+#### RBAC Interaction
+
+Like CertificateSigningRequests, it will be possible to grant RBAC permissions
+to TrustAnchorSets based on the associated signer name, rather than the name of
+the TrustAnchorSetsObject.
+
+For example, a controller that manages TrustAnchorSets would typically be
+allowed to do anything to TrustAnchorSets objects with the correct signerName:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: example-com-trustmanager
+rules:
+- apiGroups:
+  - certificates.k8s.io
+  resources:
+  - signers/trustanchorsets
+  resourceNames:
+  - example.com/my-signer-name # Or example.com/* for all signers in the example.com domain.
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - update
+  - delete
+```
+
+And a component that directly reads TrustAnchorSets (typically Kubelet) would
+probably just be allowed to read all TrustAnchorSets:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: trustanchorsets-reader
+rules:
+- apiGroups:
+  - certificates.k8s.io
+  resources:
+  - trustanchorsets
+  verbs:
+  - get
+  - list
+  - watch
+```
+
+### trustAnchorSets Projected Volume Source
+
+The trustAnchorSets projected volume source is responsible for retrieving a
+designated trust anchor set from a signer and publishing it into the container
+filesystem.
+
+#### Configuration Object Definition
```

Comment:
the API here has some interesting aspects around validation.
likely signerName must be defined (required field) and the trustAnchorSets must be len(x) > 0 and at least one TrustAnchorSetFile.FlatFile must be non-nil?

Reply:
also i know the proposed APIs are drafts, but i think it's actually useful to bake them in a close to final form in the KEP (with the proper Go-lintable comments, json tags, etc, which can ease on the API review process later.

Reply:
I think I've gotten all the requisite tags and comments in place.